### PR TITLE
ima-evm-utils-tests.py: Add ima-evm-utils test case

### PR DESCRIPTION
**Pull request by nasastry**
Add ima-evm-utils test case for distribution and upstream

Signed-off-by: Nageswara R Sastry <rnsastry@linux.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
RHEL:
JOB ID     : 1f61d8e18544f0aa236bf344925f56cf1254c492
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-28T00.20-1f61d8e/job.log
 (1/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-upstream-353a: FAIL: ima-evm-utils 'make check' failed. (13.38 s)
 (2/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-distro-7b3a: FAIL: ima-evm-utils 'make check' failed. (16.54 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-28T00.20-1f61d8e/results.html
JOB TIME   : 40.85 s

SLES:
JOB ID     : da669651372731e3efe63c2c764a9f69b7e2fb8b
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-28T10.48-da66965/job.log
 (1/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-upstream-353a: PASS (13.70 s)
 (2/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-distro-7b3a: PASS (15.67 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-28T10.48-da66965/results.html
JOB TIME   : 48.20 s

```

File: security/ima-evm-utils-tests.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        if detected_distro.name in ['rhel', 'fedora', 'centos', 'redhat']:
+            deps.extend(["keyutils-libs-devel", "libxslt", "openssl-devel",
+                         "tpm2-tss-devel"])
+        elif 'SuSE' in detected_distro.name:
+            deps.extend(['attr', 'docbook-xsl-stylesheets', 'keyutils-devel',
+                         'libattr-devel', 'libxslt-tools', 'openssl-devel',
+                         'tpm2-0-tss-devel'])
+        else:
+            self.cancel("Install the package for audit supported\
+                      by %s" % detected_distro.name)
```
Comment (Naresh-ibm) [inline security/ima-evm-utils-tests.py:45]: @nasastry I don't think cancel is required here. or we need to tweak the cancel message like distro is not supported for this test.
which package is not installed here? 
can you please re-check this?

Comment (nasastry) [reply security/ima-evm-utils-tests.py:45]: @Naresh-ibm For the other test cases also have this `cancel` .. changed the message and sending other version. Thanks!!

**Comment by nasastry** [comment]
```
RHEL:
JOB ID     : c6a6002cebb48c824d616c662024dd34b9140979
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-28T05.12-c6a6002/job.log
 (1/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-upstream-353a:  FAIL: ima-evm-utils 'make check' failed. (13.60 s)
 (2/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-distro-7b3a:  FAIL: ima-evm-utils 'make check' failed. (16.79 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-28T05.12-c6a6002/results.html
JOB TIME   : 41.57 s


SLES:
JOB ID     : f977a5d7a79c7f38ba62a804d2c694323feda5f7
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-28T15.40-f977a5d/job.log
 (1/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-upstream-353a:  PASS (17.38 s)
 (2/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-distro-7b3a:  PASS (14.88 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-28T15.40-f977a5d/results.html
JOB TIME   : 51.24 s

```

File: security/ima-evm-utils-tests.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+        run_type = self.params.get('type', default='upstream')
+        if run_type == "upstream":
+            def_url = "https://git.code.sf.net/p/linux-ima/ima-evm-utils"
+            url = self.params.get('url', default=def_url)
+            git.get_repo(url, destination_dir=self.workdir)
+            os.chdir(self.workdir)
+            process.run('./autogen.sh', ignore_status=True)
+            process.run('./configure', ignore_status=True)
```
Comment (abdhaleegit) [inline security/ima-evm-utils-tests.py:55]: do not ignore exit code as will have to exit asoonas the above commands failed

File: security/ima-evm-utils-tests.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                         'libattr-devel', 'libxslt-tools', 'openssl-devel',
+                         'tpm2-0-tss-devel'])
+        else:
+            self.cancel("%s not supported for this test" % detected_distro.name)
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+        run_type = self.params.get('type', default='upstream')
+        if run_type == "upstream":
+            def_url = "https://git.code.sf.net/p/linux-ima/ima-evm-utils"
```
Comment (abdhaleegit) [inline security/ima-evm-utils-tests.py:50]: Remove hardcoded  url here as you are getting from yaml

File: security/ima-evm-utils-tests.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            if not self.srcdir:
+                self.fail("ima-evm-utils source install failed.")
+        os.chdir(self.srcdir)
+
+    def test(self):
+        '''
+        Running tests from ima-evm-utils
+        '''
+        count = 0
+        output = build.run_make(self.srcdir, extra_args="check",
```
Comment (abdhaleegit) [inline security/ima-evm-utils-tests.py:71]: Also if build fails,, exit and  do not ignore

Comment (abdhaleegit) [reply security/ima-evm-utils-tests.py:71]: No need of below exit status check the lib has ability to fail when ignore_status is false by default

**Review by abdhaleegit — CHANGES_REQUESTED** [review]

File: security/ima-evm-utils-tests.py (hunk 2 above)
Comment (PraveenPenguin) [reply security/ima-evm-utils-tests.py:55]: I feel we need to have enhanced logic like remove ignore_status and need to handle gracefully as if any one of the commands failed as test case fail

Comment (PraveenPenguin) [reply security/ima-evm-utils-tests.py:55]: even please replicate the same in all the test cases

**Comment by nasastry** [comment]
```
SLES:
JOB ID     : 41b8173ac628809382acadbf5dbc6cc3b3fc4346
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-07T14.24-41b8173/job.log
 (1/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-upstream-353a: PASS (17.94 s)
 (2/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-distro-7b3a: PASS (17.10 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-07T14.24-41b8173/results.html
JOB TIME   : 56.94 s


RHEL:
JOB ID     : 49b6da1851cb288004ded9ca373502e7b894f8ff
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-07T03.57-49b6da1/job.log
 (1/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-upstream-353a: FAIL: ima-evm-utils 'make check' failed. (15.06 s)
 (2/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-distro-7b3a: FAIL: ima-evm-utils 'make check' failed. (18.56 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-07T03.57-49b6da1/results.html
JOB TIME   : 46.16 s
```

**Review by Naresh-ibm — APPROVED** [review]
@nasastry thanks for changes.
LGTM

File: security/ima-evm-utils-tests.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+import os
+from avocado import Test
+from avocado.utils import build, distro, git, process
+from avocado.utils.software_manager.manager import SoftwareManager
+
+
+class IMAEVMUtils(Test):
+    """
+    ima-evm-utils testsuite
+    :avocado: tags=security,testsuite
```
Comment (abdhaleegit) [inline security/ima-evm-utils-tests.py:26]: @nasastry one last suggestion,, just one or 2 lines to explain what does the test do ?

**Review by abdhaleegit — APPROVED** [review]
@nasastry apart from doc string explanation rest all is LGTM

File: security/ima-evm-utils-tests.py (hunk 5 above)
Comment (Naresh-ibm) [reply security/ima-evm-utils-tests.py:26]: @nasastry you can have a Readme.txt for more data points

**Comment by nasastry** [comment]
Added README.txt 
Attaching job.log

[sles_imaevutils_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/9817299/sles_imaevutils_job.log)
[rhel_imaevmutils_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/9817314/rhel_imaevmutils_job.log)

```
RHEL:
JOB ID     : 1d68fff7579a0484d32c5f13d31d31fb89836b61
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-19T01.01-1d68fff/job.log
 (1/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-upstream-353a: FAIL: ima-evm-utils 'make check' failed. (14.61 s)
 (2/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-distro-7b3a: FAIL: ima-evm-utils 'make check' failed. (18.14 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-19T01.01-1d68fff/results.html
JOB TIME   : 44.85 s

SLES:
JOB ID     : 60c26b66e269e901b5291d4d89ebed1a72c516a5
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-19T11.29-60c26b6/job.log
 (1/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-upstream-353a: PASS (17.44 s)
 (2/2) ima-evm-utils-tests.py:IMAEVMUtils.test;run-run_type-distro-7b3a: PASS (15.42 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-19T11.29-60c26b6/results.html
JOB TIME   : 53.79 s

```

**Pull request merged**